common/lcs/src/error.rs

```diff
@@ -11,6 +11,8 @@ pub type Result<T, E = Error> = std::result::Result<T, E>;
 pub enum Error {
     #[error("unexpected end of input")]
     Eof,
+    #[error("I/O error: {0}")]
+    Io(String),
```

Comment:
Drop the custom from impl and do the following instead, unless you only wanted to store the string and not the full io error:

```suggestion
    Io(#[from] std::io::Error),
```

Reply:
I get this compilation error:
```
binary operation `!=` cannot be applied to type `std::io::Error`
```

Reply:
ahh yes, the partialeq